esinstall/src/util.ts

```diff
@@ -56,25 +56,20 @@ export function resolveDependencyManifest(dep: string, cwd: string): [string | n
   // file path within the package with a "./package.json" instead. It's not as
   // thorough as Attempt #1, but it should work well until export maps become more
   // established & move out of experimental mode.
-  let result = [null, null] as [string | null, any | null];
-  try {
-    const fullPath = fs.realpathSync.native(require.resolve(dep, {paths: [cwd]}));
-    // Strip everything after the package name to get the package root path
-    // NOTE: This find-replace is very gross, replace with something like upath.
-    const searchPath = `${path.sep}node_modules${path.sep}${dep.replace('/', path.sep)}`;
-    const indexOfSearch = fullPath.lastIndexOf(searchPath);
-    if (indexOfSearch >= 0) {
-      const manifestPath =
-        fullPath.substring(0, indexOfSearch + searchPath.length + 1) + 'package.json';
-      result[0] = manifestPath;
-      const manifestStr = fs.readFileSync(manifestPath, {encoding: 'utf-8'});
-      result[1] = JSON.parse(manifestStr);
-    }
-  } catch (err) {
-    // ignore
-  } finally {
-    return result;
+  const result = [null, null] as [string | null, any | null];
+  const fullPath = fs.realpathSync.native(require.resolve(dep, {paths: [cwd]}));
```

Comment:
LGTM, but interestingly, it seems like Node 10 handles this differently. Is the difference in the handling of `require.resolve()` (do we need a try/catch around this)?

Reply:
looks like we uncovered an issue here where a Node v10 specific error was being swallowed. Skipped the test in v10 (since it's related to export maps, which Node v10 doesn't support)